packages/react-query/src/useBaseQuery.ts

```diff
@@ -59,6 +59,14 @@ export function useBaseQuery<
     defaultedOptions,
   )
 
+  if (process.env.NODE_ENV !== 'production') {
+    if (!defaultedOptions.queryFn) {
```

Comment:
Thoughts on also checking `enabled` in this conditional?

I often use a pattern like:
```ts
const queryKey = [ ... ];
const queryOptions = someOptional && anotherOptional
  ? { queryKey, queryFn() { ... } }
  : { queryKey, enabled: false };
```

so that I don't have to duplicate conditional checks inside the `queryFn` and `enabled`

Reply:
you want to use `skipToken` for that:

```
import { skipToken } from '@tanstack/react-query'

queryOptions = {
  queryKey,
  queryFn: someOptional && anotherOptional ? () => queryFn() : skipToken
}
```

Reply:
ohh TIL, thank you!